code/physioconv/write_event_file.py

```diff
@@ -0,0 +1,263 @@
+    }
+    color_patches = []
+
+    for trial_type in trial_types:
+        legend_color = legend_color_dict[trial_type]
+        color_patch = mpatches.Patch(color=legend_color, label=trial_type)
+        color_patches.append(color_patch)
+        events_of_type = events_df[events_df["trial-type"] == trial_type]
+
+        for index, row in events_of_type.iterrows():
+            ax1.axvline(row["onset"], color=legend_color, linestyle="--")
+            ax2.axvline(row["onset"], color=legend_color, linestyle="--")
+            ax3.axvline(row["onset"], color=legend_color, linestyle="--")
+    ax1.set_ylabel("V")
+    ax2.set_ylabel("V")
+    ax3.set_ylabel("V")
+    ax3.legend(handles=color_patches, loc="lower right")
+    ax1.set_title("RB")
+    ax2.set_title("ECG")
+    ax3.set_title("GA")
+    base_name = os.path.basename(tsv_file)
+    output_file = os.path.join(
+        output_folder, base_name.replace("_physio.tsv.gz", "_plot.png")
+    )
+    plt.tight_layout()
+    plt.savefig(output_file)
+
+
+def write_event_file(tsv_file: str) -> None:
+    """
+    This function reads the trigger channels of the physiological file, processes them to create the event file.
+
+    Parameters:
+        tsv_file (str): The path to the input gzipped TSV file containing physiological data of a task.
+
+    """
+    with gzip.open(tsv_file, "rt") as file:
+        df = pd.read_csv(file, sep="\t", header=None)
+    event_dataframe = pd.DataFrame(columns=["onset", "duration", "trial-type"])
+    if "bht" in tsv_file:
+        for index, row in df.iterrows():
+            if row[6] == 5:
+                breathin = {"onset": row[0], "duration": 2.7, "trial-type": "breath-in"}
+                breathout = {
+                    "onset": row[0] + 2.7,
+                    "duration": 2.3,
+                    "trial-type": "breath-out",
+                }
+                event_dataframe = event_dataframe.append(breathin, ignore_index=True)
+                event_dataframe = event_dataframe.append(breathout, ignore_index=True)
+
+            if row[7] == 5:
+                hold = {"onset": row[0], "duration": 15, "trial-type": "hold"}
+                event_dataframe = event_dataframe.append(hold, ignore_index=True)
+            """
+            #For the new version of the psychopy task
+            if row[6] == 5:
+                breathin = {"onset": row[0], "duration": 2.7, "trial-type": "breath-in"}
+                event_dataframe = event_dataframe.append(breathin, ignore_index=True)
+            if row[7] == 5:
+                breathout = {"onset": row[0],"duration": 2.3,"trial-type": "breath-out"}
+                event_dataframe = event_dataframe.append(breathout, ignore_index=True)
+            if row[8] == 5:
+                hold = {"onset": row[0], "duration": 2.7, "trial-type": "hold"}
+                event_dataframe = event_dataframe.append(hold, ignore_index=True)
+            """
+        json_content = {
```

Comment:
Better to avoid code repetitions. Initialize the json file above and then you just copy the dictionary here

Reply:
For each task:

```suggestion
        json_content = EVENTS_JSON_BOILERPLATE.copy()
        json_contents["StimulusPresentation"]["Code"] = "https://github.com/TheAxonLab/HCPh-fMRI-tasks/blob/97cc7879622f45129eefb9968890b41631f40851/task-bht_bold.psyexp"
        json_contents["Description"] = "Indicator of type of action that is expected"
        json_contents["LongName"] = "Breath-holding task conditions (that is, breath-in, breath-out, and hold)"
        json_contents["Levels"] = {
            "breath-in": "A green rectangle is displayed to indicate breathing in",
            "breath-out": "A yellow rectangle (orange for the last breath-in before hold) is displayed to indicate breathing out",
            "hold": "A red rectangle is displayed to indicate breath hold",
        }
```